components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -990,3 +990,36 @@ func createScriptForAddingNode(node relaxting.Node) *elastic.Script {
 	return elastic.NewScript("if (!(ctx._source.nodes instanceof Collection)) {ctx._source.nodes = [ctx._source.nodes];} ctx._source.nodes.add(params.node)").Params(params)
 
 }
+
+func (backend *ESClient) SetDayLatestToFalseForControlIndex(ctx context.Context, controlId string, profileId string, mapping mappings.Mapping, index string, nodeId string) error {
+	termQueryThisControl := elastic.NewTermsQuery("_id", GetDocIdByControlIdAndProfileID(controlId, profileId))
+
+	boolQueryDayLatest := elastic.NewBoolQuery().
+		Must(termQueryThisControl)
+
+	script := elastic.NewScript(`
+	def targets = ctx._source.nodes.findAll(node -> node.node_uuid == params.node_uuid);
+	for(node in targets) { 
+		if(node.day_latest==true) {
+			node.day_latest = false
+		}
+	}
+	if (ctx._source.day_latest==true) {
+		ctx._source.day_latest = false;
+	}`).Param("node_uuid", nodeId)
+	oneDayAgo := time.Now().Add(-24 * time.Hour)
+	time90daysAgo := time.Now().Add(-24 * time.Hour * 90)
+	// Getting all the indices
+	esIndexes, err := relaxting.IndexDates(relaxting.CompDailyControlIndexPrefix, time90daysAgo.Format(time.RFC3339), oneDayAgo.Format(time.RFC3339))
+	if err != nil {
+		logrus.Errorf("Cannot get indexes: %+v", err)
```

Comment:
we need to do error handling properly, here we are just printing the error. In this case, we need to send the error to the caller.